primitives/manta/src/currencies.rs

```diff
@@ -0,0 +1,186 @@
+// Copyright 2020-2023 Manta Network.
+// This file is part of Manta.
+//
+// Manta is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Manta is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Manta.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Manta Currencies Implementation
+
+use crate::assets::AssetConfig;
+use codec::{FullCodec, MaxEncodedLen};
+use frame_support::{
+    ensure,
+    traits::{
+        fungible, fungibles,
+        fungibles::{Mutate, Transfer},
+        Currency, ExistenceRequirement, WithdrawReasons,
+    },
+    Parameter,
+};
+use frame_system::Config;
+use orml_traits::{arithmetic::CheckedSub, MultiCurrency};
+use scale_info::TypeInfo;
+use sp_core::Get;
+use sp_runtime::{
+    traits::{AtLeast32BitUnsigned, MaybeSerializeDeserialize, Member, Zero},
+    ArithmeticError, DispatchError, DispatchResult,
+};
+use sp_std::{fmt::Debug, marker::PhantomData};
+
+/// Currencies implements for orml_traits::MultiCurrency.
+pub struct Currencies<C, A, Native, NonNative>(PhantomData<(C, A, Native, NonNative)>);
+
+impl<C, A, Native, NonNative> MultiCurrency<C::AccountId> for Currencies<C, A, Native, NonNative>
```

Comment:
isn't this super similar to the fungibleledger trait and can't we reuse that code somehow ?

Reply:
zenlink current use orml_traits::Currencies, we can't change that for now. so the only way is we implement `orml_traits::Currencies`.